src/github/client.py

```diff
@@ -5,11 +5,9 @@
 from src.github.get_app_token import sgtm_github_auth
 from src.logger import logger
 
-gh_client = sgtm_github_auth.get_rest_client()
-
 
 def _get_pull_request(owner: str, repository: str, number: int) -> PullRequest:  # type: ignore
-    repo = gh_client.get_repo(f"{owner}/{repository}")
+    repo = sgtm_github_auth(owner).get_rest_client().get_repo(f"{owner}/{repository}")
```

Comment:
oh hmm does this mean we're creating a new client for every request now

Reply:
no the `sgtm_github_auth` method accesses a map of `github_org_name -> github_auth_client` to create and/or retrieve an existing client to use.

Reply:
oh actually, we are creating a new auth for every test run so I'll make this more robust for local auth